kafka-rest/src/main/java/io/confluent/kafkarest/ConsumerReadTask.java

```diff
@@ -115,13 +128,18 @@ public void doPartialRead() {
           ConsumerRecordAndSize<ClientKeyT, ClientValueT> recordAndSize =
                   parent.createConsumerRecord(msg);
           roughMsgSize = recordAndSize.getSize();
-          if (bytesConsumed + roughMsgSize >= maxResponseBytes) {
+          this.willExceedMaxResponseBytes = bytesConsumed + roughMsgSize >= maxResponseBytes;
```

Comment:
This value is recomputed on every call to `doPartialRead` and it is only used from within `doPartialRead`. Couldn't it be a local variable instead of a field? Same for `exceededMinResponseBytes`.

Reply:
https://github.com/confluentinc/kafka-rest/blob/7ce62592ba58bf4cf71e176146ea68f7555d63be/kafka-rest/src/main/java/io/confluent/kafkarest/ConsumerReadTask.java#L97

We assign them when inheriting a previous task which might have failed. I think the problem is that we do not respect those in that case.

Reply:
I guess my point is that we can always recompute this using `bytesConsumed`, so why do we need the additional fields?